src/main/java/com/easypost/model/Shipment.java

```diff
@@ -1028,6 +1071,39 @@ public Shipment buy(final Map<String, Object> params, final String apiKey) throw
      */
     public Shipment buy(final Map<String, Object> params, final boolean withCarbonOffset, final String apiKey)
             throws EasyPostException {
+        return this.buy(params, withCarbonOffset, null, apiKey);
+    }
+
+    /**
+     * Buy this Shipment.
+     *
+     * @param params           the options for the query.
+     * @param endShipperId     the id of the end shipper to use for this purchase.
+     * @param apiKey           API key to use in request (overrides default API key).
+     * @return Shipment object
+     * @throws EasyPostException when the request fails.
+     */
+    public Shipment buy(final Map<String, Object> params, final String endShipperId, final String apiKey)
+            throws EasyPostException {
+        return this.buy(params, false, endShipperId, apiKey);
+    }
+
+    /**
+     * Buy this Shipment.
+     *
+     * @param params           the options for the query.
+     * @param withCarbonOffset whether to include a carbon offset when buying the shipment.
+     * @param endShipperId     the id of the end shipper to use for this purchase.
+     * @param apiKey           API key to use in request (overrides default API key).
+     * @return Shipment object
+     * @throws EasyPostException when the request fails.
+     */
+    public Shipment buy(final Map<String, Object> params, final boolean withCarbonOffset, 
+        final String endShipperId, final String apiKey) throws EasyPostException {
+        if (endShipperId != null && !endShipperId.isEmpty()) {
+            params.put("end_shipper_id", endShipperId);
+        }
```

Comment:
Issue: Order of operations here, this should come after carbon offset since that param comes before end shipper.

Reply:
fixed